packages/vendure-plugin-shipping-extensions/src/distance-based-shipping-calculator.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  LanguageCode,
+  ShippingCalculator,
+  Injector,
+  InternalServerError,
+} from '@vendure/core';
+import { ShippingExtensionsOptions } from './shipping-extensions.plugin';
+import { PLUGIN_OPTIONS } from './constants';
+import { getDistanceBetweenPointsInKMs } from './get-distance-between-points';
+
+let pluginOptions: ShippingExtensionsOptions;
+export const distanceBasedShippingCalculator = new ShippingCalculator({
+  code: 'distance-based-shipping-calculator',
+  description: [
+    {
+      languageCode: LanguageCode.en,
+      value: 'Distance Based Shipping Calculator',
+    },
+  ],
+  args: {
+    storeLatitude: {
+      type: 'float',
+      ui: { component: 'number-form-input', min: -90, max: 90 },
+      label: [{ languageCode: LanguageCode.en, value: 'Store Latitude' }],
+    },
+    storeLongitude: {
+      type: 'float',
+      ui: { component: 'number-form-input', min: -180, max: 180 },
+      label: [{ languageCode: LanguageCode.en, value: 'Store Longitude' }],
+    },
+    pricePerKm: {
+      type: 'float',
+      ui: { component: 'currency-form-input' },
+      label: [{ languageCode: LanguageCode.en, value: 'Price per KM' }],
+    },
+    taxRate: {
+      type: 'float',
+      ui: { component: 'number-form-input', suffix: '%', min: 0, max: 100 },
+      label: [{ languageCode: LanguageCode.en, value: 'Tax rate' }],
+    },
+  },
+  init(injector: Injector) {
+    pluginOptions = injector.get<ShippingExtensionsOptions>(PLUGIN_OPTIONS);
+  },
+  calculate: async (ctx, order, args) => {
```

Comment:
I think we need to handle the scenario where an `order.shippingAddress` isn't set yet. You could do `eligibleShippingMethods` query even for orders without shippingaddress right? I think it's fine to return $0.

Do you think we need this?

Reply:
I think the other option we have is let the configured `OrderAddressToGeolocationConversionStrategy` handle this scenario, but I think returning 0 makes more sense than that @martijnvdbrug